build.sbt

```diff
@@ -1,22 +1,105 @@
+import xerial.sbt.Sonatype._
+
 import com.scalapenos.sbt.prompt.SbtPrompt.autoImport._
 
 promptTheme := com.scalapenos.sbt.prompt.PromptThemes.ScalapenosTheme
 
-/** Project configurations */
+val commonSettings = Seq(
+  // sbt-git provides the isSnapshot.value variable, which will return true if there
+  // are no tags associated with the HEAD commit, or if there are uncommitted
+  // changes. However, sbt-git only appends the -SNAPSHOT suffix if there are
+  // uncommitted changes in the workspace.
+  //
+  // https://github.com/sbt/sbt-git/blob/f8caf9365be380cf101e9605af159b5e7f842d0c/src/main/scala/com/typesafe/sbt/SbtGit.scala#L173
+  version := {
+    // Avoid Cyclic reference involving error
+    if (git.gitCurrentTags.value.isEmpty || git.gitUncommittedChanges.value)
+      git.gitDescribedVersion.value.get + "-SNAPSHOT"
+    else
+      git.gitDescribedVersion.value.get
+  },
+  scalaVersion := "2.11.12",
+  crossScalaVersions := Seq("2.11.12", "2.12.8"),
+  resolvers ++= Seq(
+    Resolver.sonatypeRepo("releases"),
+    "locationtech-releases" at "https://repo.locationtech.org/content/groups/releases",
+    "locationtech-snapshots" at "https://repo.locationtech.org/content/groups/snapshots"
+  ),
+  addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
+  scalacOptions := Seq(
+    "-deprecation",
+    "-unchecked",
+    "-feature",
+    "-language:implicitConversions",
+    "-language:reflectiveCalls",
+    "-language:higherKinds",
+    "-language:postfixOps",
+    "-language:existentials",
+    "-language:experimental.macros",
+    "-feature",
+    "-Ypatmat-exhaust-depth", "100"
+  )
+)
+
+lazy val noPublishSettings = Seq(
+  publish := {},
+  publishLocal := {},
+  publishArtifact := false
+)
+
+lazy val publishSettings = Seq(
+  organization := "com.azavea.geotrellis",
```

Comment:
Artifacts will live on Maven Central under `com/azavea/geotrellis/maml-jvm_2.12/`, etc. I think this is fair.